packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarouselNav/useTeachingPopoverCarouselNav.tsx

```diff
@@ -26,16 +27,18 @@ export const useTeachingPopoverCarouselNav_unstable = (
     unstable_hasDefault: true,
   });
 
-  const totalPages = useTeachingPopoverCarouselContext_unstable(context => context.totalPages);
-  const currentPage = useTeachingPopoverCarouselContext_unstable(context => context.currentPage);
+  const currentPage = useCarouselContext_unstable(context => context.currentIndex);
+
+  const store = useCarouselContext_unstable(context => context.store);
+  const values = useSyncExternalStore(store.subscribe, () => store.getSnapshot());
 
   // Generate the child TeachingPopoverCarouselNavButton and memoize them to prevent unnecessary re-rendering
   const rootChildren = React.useMemo(() => {
-    return Array.from(Array(totalPages), (_, i) => <TeachingPopoverCarouselNavButton key={i} index={i} />);
-  }, [totalPages]);
+    return Array.from(Array(values.length), (_, i) => <TeachingPopoverCarouselNavButton key={i} index={i} />);
+  }, [values.length]);
 
   return {
-    totalPages,
+    totalPages: values.length,
```

Comment:
nit: let's make this a bit more readable

```ts
  const store = useCarouselContext_unstable(context => context.store);
  const currentPage = useCarouselContext_unstable(context => context.currentIndex);

  const values = useSyncExternalStore(store.subscribe, () => store.getSnapshot());
  const totalPages = values.length;

  // Generate the child TeachingPopoverCarouselNavButton and memoize them to prevent unnecessary re-rendering
  const rootChildren = React.useMemo(() => {
    return Array.from(Array(totalPages), (_, i) => <TeachingPopoverCarouselNavButton key={i} index={i} />);
  }, [totalPages]);

  return {
    totalPages: totalPages,
```